.github/workflows/testing.yml

```diff
@@ -0,0 +1,223 @@
+name: Automated Tests
+
+on: [push]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+    - name: Install flake8
+      shell: bash
+      run: |
+        python -V
+        python -m pip install flake8
+    - name: Run flake8 on diff with upstream/master
+      shell: bash
+      run: |
+        bash ./.github/scripts/flake8_diff.sh
+
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python_version: [2.7, 3.5, 3.6, 3.7, 3.8, "pypy3"]
+        exclude:
+          # Do not test all minor version on all platforms, especially if they
+          # are not the oldest/newest supported versions
+          - os: windows-latest
+            python_version: 3.6
+          - os: windows-latest
+            python_version: 3.7
+          - os: windows-latest
```

Comment:
I am failing to build some dependencies on`pypy` and `Windows` - this is why I am skipping this config. I should add a comment.